tee-worker/omni-executor/aa-contracts-client/src/entry_point_client.rs

```diff
@@ -0,0 +1,547 @@
+		let tx = build_call_transaction(self.entry_point_address, call_data);
+		let result = self.rpc_client.call(tx).await.map_err(|_| error!("Could not send tx"))?;
+		let hash: FixedBytes<32> = FixedBytes::abi_decode(&result).unwrap();
+		Ok(hash)
+	}
+
+	#[allow(clippy::too_many_arguments)]
+	pub async fn create_packed_user_operation(
+		&self,
+		factory_address: Address,
+		oa: [u8; 32],
+		client_id: &[u8],
+		root_address: Address,
+		call_data: Bytes,
+		paymaster_address: Option<Address>,
+	) -> Result<PackedUserOperation, ()> {
+		// Create init code using existing helper
+		let init_code_bytes =
+			prepare_factory_init_code(factory_address, oa, client_id, root_address);
+		let init_code = Bytes::from(init_code_bytes);
+
+		// Get sender address from EntryPoint
+		let sender = self.get_sender_address(init_code.clone()).await?;
+
+		// Check if the account already has code deployed
+		let code = self.rpc_client.get_code_at(sender).await.map_err(|_| ())?;
+
+		// Get nonce - if smart wallet doesn't exist yet, use 0
+		let nonce = if code.is_empty() {
+			// Smart wallet doesn't exist yet, use 0 as nonce
+			U256::from(0)
+		} else {
+			// Smart wallet exists, get nonce from contract
+			let smart_wallet_client = SmartWalletClient::new(sender, self.rpc_client.clone());
+			smart_wallet_client.get_nonce().await?
+		};
+		let init_code_to_use = if code.is_empty() {
+			// No code at address, include init code
+			init_code
+		} else {
+			// Code already exists, no init code needed
+			Bytes::new()
+		};
+
+		// Default gas limits - can be adjusted based on requirements
+		let verification_gas_limit = U256::from(250000u64); // Higher for verification
+		let call_gas_limit = U256::from(50000u64); // Lower for simple calls
+		let account_gas_limits = create_account_gas_limits(verification_gas_limit, call_gas_limit);
+
+		let pre_verification_gas = U256::from(21000u64);
+		let max_fee_per_gas = U256::from(3000000000u64); // 3 gwei
+		let max_priority_fee_per_gas = U256::from(1000000000u64); // 1 gwei
+		let gas_fees = create_gas_fees(max_fee_per_gas, max_priority_fee_per_gas);
+
+		let paymaster_and_data = if let Some(paymaster_addr) = paymaster_address {
+			create_paymaster_and_data(paymaster_addr, U256::from(50000u64), U256::from(50000u64))
+		} else {
+			Bytes::new()
+		};
+
+		Ok(PackedUserOperation {
+			sender,
+			nonce,
+			initCode: init_code_to_use,
+			callData: call_data,
+			accountGasLimits: account_gas_limits,
+			preVerificationGas: pre_verification_gas,
+			gasFees: gas_fees,
+			paymasterAndData: paymaster_and_data,
+			sessionAccount: Address::default(),
+			sessionExpiration: U256::from(0),
+			sessionAccountProof: Bytes::new(),
+			signature: Bytes::new(),
+		})
+	}
+
+	/// Create PackedUserOperation using local CREATE2 address calculation
+	/// This is more efficient as it avoids the EntryPoint.getSenderAddress RPC call
+	#[allow(clippy::too_many_arguments)]
+	pub async fn create_packed_user_operation_with_local_address(
```

Comment:
Later we should be able to extract common code of the two fn

Reply:
extracted